### ISSUE TITLE
CFamily: enable reporting parsing errors as issues

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonarlint-visualstudio#1109]
### Description

By default the CFamily analyser does not report parsing errors as issues. There is a hidden rule that will report parsing error issues but it is enabled by default, and is not listed in `rules.sonarsource.com`.

However, the rule can be useful for diagnosing FPs/FNs/unexpected analysis results so it would be useful if there was a simple way for the user to turn the rule on and off e.g. via `Tools`, `Options`, or by listing the rule in the default `settings.json` file.

![image](https://user-images.githubusercontent.com/10757559/69532058-559a7e80-0f75-11ea-99f6-9aab94ac11a8.png)

### Workaround
Manually enable the rule in `settings.json`:

```
{
  "sonarlint.rules": {
    "cpp:ParsingError": {
      "Level": "On"
    },
  "sonarlint.rules": {
    "c:ParsingError": {
      "Level": "On"
    }
  }
}
```

### Additional information
SonarLint version: v4.13+
Languages: C, C++

**Comment by duncanp-sonar** [comment]
Not required if we fix #1295
